crates/generate/src/types/struct.rs

```diff
@@ -10,87 +10,9 @@ pub(super) fn define_struct(
     name: &witx::Id,
     s: &witx::StructDatatype,
 ) -> TokenStream {
-    if !s.needs_lifetime() {
-        define_copy_struct(names, name, s)
-    } else {
-        define_ptr_struct(names, name, s)
-    }
-}
-
-fn define_copy_struct(names: &Names, name: &witx::Id, s: &witx::StructDatatype) -> TokenStream {
     let ident = names.type_(name);
     let size = s.mem_size_align().size as u32;
-    let align = s.mem_size_align().align as u32;
-
-    let member_decls = s.members.iter().map(|m| {
-        let name = names.struct_member(&m.name);
-        let type_ = names.type_ref(&m.tref, anon_lifetime());
-        quote!(pub #name: #type_)
-    });
-    let member_valids = s.member_layout().into_iter().map(|ml| {
-        let type_ = names.type_ref(&ml.member.tref, anon_lifetime());
-        let offset = ml.offset as u32;
-        let fieldname = names.struct_member(&ml.member.name);
-        quote! {
-            #type_::validate(
-                &ptr.cast(#offset).map_err(|e|
-                    wiggle_runtime::GuestError::InDataField{
-                        typename: stringify!(#ident).to_owned(),
-                        field: stringify!(#fieldname).to_owned(),
-                        err: Box::new(e),
-                    })?
-                ).map_err(|e|
-                    wiggle_runtime::GuestError::InDataField {
-                        typename: stringify!(#ident).to_owned(),
-                        field: stringify!(#fieldname).to_owned(),
-                        err: Box::new(e),
-                    })?;
-        }
-    });
-
-    quote! {
-        #[repr(C)]
-        #[derive(Copy, Clone, Debug, PartialEq)]
-        pub struct #ident {
-            #(#member_decls),*
-        }
-
-        impl<'a> wiggle_runtime::GuestType<'a> for #ident {
-            fn size() -> u32 {
-                #size
-            }
-
-            fn align() -> u32 {
-                #align
-            }
-
-            fn name() -> String {
-                stringify!(#ident).to_owned()
-            }
-
-            fn validate(ptr: &wiggle_runtime::GuestPtr<#ident>) -> Result<(), wiggle_runtime::GuestError> {
-                #(#member_valids)*
-                Ok(())
-            }
-
-            fn read(location: &wiggle_runtime::GuestPtr<'a, #ident>) -> Result<#ident, wiggle_runtime::GuestError> {
-                let r = location.as_ref()?;
-                Ok(*r)
-            }
-
-            fn write(&self, location: &wiggle_runtime::GuestPtrMut<'a, Self>) {
-                unsafe { (location.as_raw() as *mut #ident).write(*self) };
-            }
-        }
-
-        impl<'a> wiggle_runtime::GuestTypeTransparent<'a> for #ident {}
-    }
-}
-
-fn define_ptr_struct(names: &Names, name: &witx::Id, s: &witx::StructDatatype) -> TokenStream {
-    let ident = names.type_(name);
-    let size = s.mem_size_align().size as u32;
-    let align = s.mem_size_align().align as u32;
+    let align = s.mem_size_align().align as usize;
```

Comment:
Hmm, shouldn't this be `u32` and not `usize`?

Reply:
Mostly just convenience, it never really needed to be a `u32` so I went ahead and made alignment a `usize` everywhere to help reduce casts (in theory)

Reply:
Got ya! Makes sense!